# Homework 4 - Data Preparation

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('flights.csv')
df_airports = pd.read_csv('airports.csv')
df_airlines = pd.read_csv('airlines.csv')

In [3]:
print(df.shape)

(1500000, 32)


In [4]:
df.sample(4)

Unnamed: 0  YEAR  MONTH  DAY  DAY_OF_WEEK AIRLINE  FLIGHT_NUMBER  \
489216       391714  2015      1   26            1      WN           4298   
879898      4649362  2015     10   17            6      OO           2869   
1416121     5556279  2015     12   15            2      EV           4327   
660027      4695705  2015     10   20            2      OO           4609   

        TAIL_NUMBER ORIGIN_AIRPORT DESTINATION_AIRPORT  ...  ARRIVAL_TIME  \
489216       N907WN            BUR                 SJC  ...        1834.0   
879898       N492SW          10140               12892  ...        1627.0   
1416121      N17560            EWR                 IND  ...         922.0   
660027       N614SK          10713               14869  ...        1621.0   

         ARRIVAL_DELAY  DIVERTED  CANCELLED  CANCELLATION_REASON  \
489216           -16.0         0          0                  NaN   
879898           -10.0         0          0                  NaN   
1416121            7.0         0          0                  NaN   
660027             2.0         0          0                  NaN   

         AIR_SYSTEM_DELAY  SECURITY_DELAY  AIRLINE_DELAY  LATE_AIRCRAFT_DELAY  \
489216                NaN             NaN            NaN                  NaN   
879898                NaN             NaN            NaN                  NaN   
1416121               NaN             NaN            NaN                  NaN   
660027                NaN             NaN            NaN                  NaN   

         WEATHER_DELAY  
489216             NaN  
879898             NaN  
1416121            NaN  
660027             NaN  

[4 rows x 32 columns]

In [5]:
df.columns

Index(['Unnamed: 0', 'YEAR', 'MONTH', 'DAY', 'DAY_OF_WEEK', 'AIRLINE',
       'FLIGHT_NUMBER', 'TAIL_NUMBER', 'ORIGIN_AIRPORT', 'DESTINATION_AIRPORT',
       'SCHEDULED_DEPARTURE', 'DEPARTURE_TIME', 'DEPARTURE_DELAY', 'TAXI_OUT',
       'WHEELS_OFF', 'SCHEDULED_TIME', 'ELAPSED_TIME', 'AIR_TIME', 'DISTANCE',
       'WHEELS_ON', 'TAXI_IN', 'SCHEDULED_ARRIVAL', 'ARRIVAL_TIME',
       'ARRIVAL_DELAY', 'DIVERTED', 'CANCELLED', 'CANCELLATION_REASON',
       'AIR_SYSTEM_DELAY', 'SECURITY_DELAY', 'AIRLINE_DELAY',
       'LATE_AIRCRAFT_DELAY', 'WEATHER_DELAY'],
      dtype='object')

#### **Q1** In the flights dataset, which is the column with the most number of null values?

In [6]:
df.isnull().sum().sort_values(ascending=False)

CANCELLATION_REASON    1476856
WEATHER_DELAY          1225838
LATE_AIRCRAFT_DELAY    1225838
AIRLINE_DELAY          1225838
SECURITY_DELAY         1225838
AIR_SYSTEM_DELAY       1225838
AIR_TIME                 26963
ARRIVAL_DELAY            26963
ELAPSED_TIME             26963
TAXI_IN                  23808
WHEELS_ON                23808
ARRIVAL_TIME             23808
TAXI_OUT                 22955
WHEELS_OFF               22955
DEPARTURE_TIME           22237
DEPARTURE_DELAY          22237
TAIL_NUMBER               3818
SCHEDULED_TIME               1
CANCELLED                    0
MONTH                        0
DAY                          0
DAY_OF_WEEK                  0
AIRLINE                      0
FLIGHT_NUMBER                0
DESTINATION_AIRPORT          0
DIVERTED                     0
SCHEDULED_ARRIVAL            0
SCHEDULED_DEPARTURE          0
DISTANCE                     0
ORIGIN_AIRPORT               0
YEAR                         0
Unnamed: 0                   0
dtype: i

In [7]:
# We can look up the list above, or use the below to list the exact column name:
df.isnull().sum().sort_values(ascending=False).index[0]

'CANCELLATION_REASON'

#### **Q2** How many unique destinations are there in the flights dataset?

In [8]:
df.DESTINATION_AIRPORT.nunique()

627

#### **Q3** How many unique tail numbers are associated with flights operated by American Airlines (AA)?

In [9]:
df[df.AIRLINE=='AA'].TAIL_NUMBER.nunique()

1036

#### **Q4** Among the airlines listed in the dataset, which one has the highest average arrival delay?

In [10]:
df.groupby('AIRLINE').ARRIVAL_DELAY.mean().sort_values(ascending=False)

AIRLINE
NK    14.053750
F9    12.211814
B6     6.620890
EV     6.537097
MQ     6.351351
OO     5.820780
UA     5.428603
WN     4.324226
VX     4.261044
US     3.545975
AA     3.459706
HA     1.889286
DL     0.290981
AS    -0.952565
Name: ARRIVAL_DELAY, dtype: float64

#### **Q5** In the flights dataset, how many complete rows are left if we remove all observations that have any missing value?

In [11]:
df.dropna().shape

(0, 32)

In [12]:
len(df.dropna())

0

#### **Q6** Create a query for the flights dataset where either of the following conditions are true:
#### 1. Airline is American Airlines (AA), and ORIGIN_AIRPORT is JFK.
#### 2. The airline is Virgin America
#### How many rows are returned by the query?

In [13]:
# note that the `|` symbol is the operator OR
df[((df['AIRLINE'] == 'AA') & (df['ORIGIN_AIRPORT'] == 'JFK')) | (df['AIRLINE'] == 'VX')] 

Unnamed: 0  YEAR  MONTH  DAY  DAY_OF_WEEK AIRLINE  FLIGHT_NUMBER  \
55          5042222  2015     11   11            3      VX            919   
194         2683425  2015      6   18            4      AA             65   
283          317590  2015      1   21            3      AA            235   
318         5681558  2015     12   22            2      AA            235   
394         4984587  2015     11    8            7      VX           1930   
...             ...   ...    ...  ...          ...     ...            ...   
1499854     5324762  2015     11   30            1      VX            399   
1499886     5137257  2015     11   17            2      VX            347   
1499902      131285  2015      1    9            5      VX            232   
1499928     3225311  2015      7   21            2      AA            936   
1499979     1728688  2015      4   21            2      VX            501   

        TAIL_NUMBER ORIGIN_AIRPORT DESTINATION_AIRPORT  ...  ARRIVAL_TIME  \
55           N524VA            LAS                 SFO  ...        1837.0   
194          N867AA            JFK                 MIA  ...        2334.0   
283          N3LYAA            JFK                 SEA  ...        2103.0   
318          N3CGAA            JFK                 BOS  ...        2209.0   
394          N855VA            SFO                 LAX  ...         932.0   
...             ...            ...                 ...  ...           ...   
1499854      N637VA            JFK                 LAX  ...         957.0   
1499886      N840VA            FLL                 SFO  ...        2048.0   
1499902      N853VA            LAX                 ORD  ...        1704.0   
1499928      N5EWAA            JFK                 STT  ...        1546.0   
1499979      N632VA            JFK                 FLL  ...        1043.0   

         ARRIVAL_DELAY  DIVERTED  CANCELLED  CANCELLATION_REASON  \
55                 7.0         0          0                  NaN   
194              181.0         0          0                  NaN   
283              -56.0         0          0                  NaN   
318               24.0         0          0                  NaN   
394               12.0         0          0                  NaN   
...                ...       ...        ...                  ...   
1499854          -28.0         0          0                  NaN   
1499886          -17.0         0          0                  NaN   
1499902           24.0         0          0                  NaN   
1499928          212.0         0          0                  NaN   
1499979          -22.0         0          0                  NaN   

         AIR_SYSTEM_DELAY  SECURITY_DELAY  AIRLINE_DELAY  LATE_AIRCRAFT_DELAY  \
55                    NaN             NaN            NaN                  NaN   
194                 181.0             0.0            0.0                  0.0   
283                   NaN             NaN            NaN                  NaN   
318                   0.0             0.0            0.0                 24.0   
394                   NaN             NaN            NaN                  NaN   
...                   ...             ...            ...                  ...   
1499854               NaN             NaN            NaN                  NaN   
1499886               NaN             NaN            NaN                  NaN   
1499902               0.0             0.0           24.0                  0.0   
1499928               0.0             0.0            0.0                212.0   
1499979               NaN             NaN            NaN                  NaN   

         WEATHER_DELAY  
55                 NaN  
194                0.0  
283                NaN  
318                0.0  
394                NaN  
...                ...  
1499854            NaN  
1499886            NaN  
1499902            0.0  
1499928            0.0  
1499979            NaN  

[19853 rows x 32 columns]

In [14]:
df[((df['AIRLINE'] == 'AA') & (df['ORIGIN_AIRPORT'] == 'JFK')) | (df['AIRLINE'] == 'VX')] 

Unnamed: 0  YEAR  MONTH  DAY  DAY_OF_WEEK AIRLINE  FLIGHT_NUMBER  \
55          5042222  2015     11   11            3      VX            919   
194         2683425  2015      6   18            4      AA             65   
283          317590  2015      1   21            3      AA            235   
318         5681558  2015     12   22            2      AA            235   
394         4984587  2015     11    8            7      VX           1930   
...             ...   ...    ...  ...          ...     ...            ...   
1499854     5324762  2015     11   30            1      VX            399   
1499886     5137257  2015     11   17            2      VX            347   
1499902      131285  2015      1    9            5      VX            232   
1499928     3225311  2015      7   21            2      AA            936   
1499979     1728688  2015      4   21            2      VX            501   

        TAIL_NUMBER ORIGIN_AIRPORT DESTINATION_AIRPORT  ...  ARRIVAL_TIME  \
55           N524VA            LAS                 SFO  ...        1837.0   
194          N867AA            JFK                 MIA  ...        2334.0   
283          N3LYAA            JFK                 SEA  ...        2103.0   
318          N3CGAA            JFK                 BOS  ...        2209.0   
394          N855VA            SFO                 LAX  ...         932.0   
...             ...            ...                 ...  ...           ...   
1499854      N637VA            JFK                 LAX  ...         957.0   
1499886      N840VA            FLL                 SFO  ...        2048.0   
1499902      N853VA            LAX                 ORD  ...        1704.0   
1499928      N5EWAA            JFK                 STT  ...        1546.0   
1499979      N632VA            JFK                 FLL  ...        1043.0   

         ARRIVAL_DELAY  DIVERTED  CANCELLED  CANCELLATION_REASON  \
55                 7.0         0          0                  NaN   
194              181.0         0          0                  NaN   
283              -56.0         0          0                  NaN   
318               24.0         0          0                  NaN   
394               12.0         0          0                  NaN   
...                ...       ...        ...                  ...   
1499854          -28.0         0          0                  NaN   
1499886          -17.0         0          0                  NaN   
1499902           24.0         0          0                  NaN   
1499928          212.0         0          0                  NaN   
1499979          -22.0         0          0                  NaN   

         AIR_SYSTEM_DELAY  SECURITY_DELAY  AIRLINE_DELAY  LATE_AIRCRAFT_DELAY  \
55                    NaN             NaN            NaN                  NaN   
194                 181.0             0.0            0.0                  0.0   
283                   NaN             NaN            NaN                  NaN   
318                   0.0             0.0            0.0                 24.0   
394                   NaN             NaN            NaN                  NaN   
...                   ...             ...            ...                  ...   
1499854               NaN             NaN            NaN                  NaN   
1499886               NaN             NaN            NaN                  NaN   
1499902               0.0             0.0           24.0                  0.0   
1499928               0.0             0.0            0.0                212.0   
1499979               NaN             NaN            NaN                  NaN   

         WEATHER_DELAY  
55                 NaN  
194                0.0  
283                NaN  
318                0.0  
394                NaN  
...                ...  
1499854            NaN  
1499886            NaN  
1499902            0.0  
1499928            0.0  
1499979            NaN  

[19853 rows x 32 columns]

In [15]:
# or we can just ask for the length of the resulting dataframe to get the answer directly
len(df[((df['AIRLINE'] == 'AA') & (df['ORIGIN_AIRPORT'] == 'JFK')) | (df['AIRLINE'] == 'VX')] )

19853

In [16]:
flights = df
flights.query('AIRLINE == "AA" and ORIGIN_AIRPORT == "JFK" | AIRLINE == "VX"')

Unnamed: 0  YEAR  MONTH  DAY  DAY_OF_WEEK AIRLINE  FLIGHT_NUMBER  \
55          5042222  2015     11   11            3      VX            919   
194         2683425  2015      6   18            4      AA             65   
283          317590  2015      1   21            3      AA            235   
318         5681558  2015     12   22            2      AA            235   
394         4984587  2015     11    8            7      VX           1930   
...             ...   ...    ...  ...          ...     ...            ...   
1499854     5324762  2015     11   30            1      VX            399   
1499886     5137257  2015     11   17            2      VX            347   
1499902      131285  2015      1    9            5      VX            232   
1499928     3225311  2015      7   21            2      AA            936   
1499979     1728688  2015      4   21            2      VX            501   

        TAIL_NUMBER ORIGIN_AIRPORT DESTINATION_AIRPORT  ...  ARRIVAL_TIME  \
55           N524VA            LAS                 SFO  ...        1837.0   
194          N867AA            JFK                 MIA  ...        2334.0   
283          N3LYAA            JFK                 SEA  ...        2103.0   
318          N3CGAA            JFK                 BOS  ...        2209.0   
394          N855VA            SFO                 LAX  ...         932.0   
...             ...            ...                 ...  ...           ...   
1499854      N637VA            JFK                 LAX  ...         957.0   
1499886      N840VA            FLL                 SFO  ...        2048.0   
1499902      N853VA            LAX                 ORD  ...        1704.0   
1499928      N5EWAA            JFK                 STT  ...        1546.0   
1499979      N632VA            JFK                 FLL  ...        1043.0   

         ARRIVAL_DELAY  DIVERTED  CANCELLED  CANCELLATION_REASON  \
55                 7.0         0          0                  NaN   
194              181.0         0          0                  NaN   
283              -56.0         0          0                  NaN   
318               24.0         0          0                  NaN   
394               12.0         0          0                  NaN   
...                ...       ...        ...                  ...   
1499854          -28.0         0          0                  NaN   
1499886          -17.0         0          0                  NaN   
1499902           24.0         0          0                  NaN   
1499928          212.0         0          0                  NaN   
1499979          -22.0         0          0                  NaN   

         AIR_SYSTEM_DELAY  SECURITY_DELAY  AIRLINE_DELAY  LATE_AIRCRAFT_DELAY  \
55                    NaN             NaN            NaN                  NaN   
194                 181.0             0.0            0.0                  0.0   
283                   NaN             NaN            NaN                  NaN   
318                   0.0             0.0            0.0                 24.0   
394                   NaN             NaN            NaN                  NaN   
...                   ...             ...            ...                  ...   
1499854               NaN             NaN            NaN                  NaN   
1499886               NaN             NaN            NaN                  NaN   
1499902               0.0             0.0           24.0                  0.0   
1499928               0.0             0.0            0.0                212.0   
1499979               NaN             NaN            NaN                  NaN   

         WEATHER_DELAY  
55                 NaN  
194                0.0  
283                NaN  
318                0.0  
394                NaN  
...                ...  
1499854            NaN  
1499886            NaN  
1499902            0.0  
1499928            0.0  
1499979            NaN  

[19853 rows x 32 columns]

In [17]:
# flights.query(('AIRLINE == "AA" and ORIGIN_AIRPORT == "JFK"'))
flights.query('(AIRLINE == "AA" and ORIGIN_AIRPORT == "JFK") or (AIRLINE == "VX")' or something and nothing)

Unnamed: 0  YEAR  MONTH  DAY  DAY_OF_WEEK AIRLINE  FLIGHT_NUMBER  \
55          5042222  2015     11   11            3      VX            919   
194         2683425  2015      6   18            4      AA             65   
283          317590  2015      1   21            3      AA            235   
318         5681558  2015     12   22            2      AA            235   
394         4984587  2015     11    8            7      VX           1930   
...             ...   ...    ...  ...          ...     ...            ...   
1499854     5324762  2015     11   30            1      VX            399   
1499886     5137257  2015     11   17            2      VX            347   
1499902      131285  2015      1    9            5      VX            232   
1499928     3225311  2015      7   21            2      AA            936   
1499979     1728688  2015      4   21            2      VX            501   

        TAIL_NUMBER ORIGIN_AIRPORT DESTINATION_AIRPORT  ...  ARRIVAL_TIME  \
55           N524VA            LAS                 SFO  ...        1837.0   
194          N867AA            JFK                 MIA  ...        2334.0   
283          N3LYAA            JFK                 SEA  ...        2103.0   
318          N3CGAA            JFK                 BOS  ...        2209.0   
394          N855VA            SFO                 LAX  ...         932.0   
...             ...            ...                 ...  ...           ...   
1499854      N637VA            JFK                 LAX  ...         957.0   
1499886      N840VA            FLL                 SFO  ...        2048.0   
1499902      N853VA            LAX                 ORD  ...        1704.0   
1499928      N5EWAA            JFK                 STT  ...        1546.0   
1499979      N632VA            JFK                 FLL  ...        1043.0   

         ARRIVAL_DELAY  DIVERTED  CANCELLED  CANCELLATION_REASON  \
55                 7.0         0          0                  NaN   
194              181.0         0          0                  NaN   
283              -56.0         0          0                  NaN   
318               24.0         0          0                  NaN   
394               12.0         0          0                  NaN   
...                ...       ...        ...                  ...   
1499854          -28.0         0          0                  NaN   
1499886          -17.0         0          0                  NaN   
1499902           24.0         0          0                  NaN   
1499928          212.0         0          0                  NaN   
1499979          -22.0         0          0                  NaN   

         AIR_SYSTEM_DELAY  SECURITY_DELAY  AIRLINE_DELAY  LATE_AIRCRAFT_DELAY  \
55                    NaN             NaN            NaN                  NaN   
194                 181.0             0.0            0.0                  0.0   
283                   NaN             NaN            NaN                  NaN   
318                   0.0             0.0            0.0                 24.0   
394                   NaN             NaN            NaN                  NaN   
...                   ...             ...            ...                  ...   
1499854               NaN             NaN            NaN                  NaN   
1499886               NaN             NaN            NaN                  NaN   
1499902               0.0             0.0           24.0                  0.0   
1499928               0.0             0.0            0.0                212.0   
1499979               NaN             NaN            NaN                  NaN   

         WEATHER_DELAY  
55                 NaN  
194                0.0  
283                NaN  
318                0.0  
394                NaN  
...                ...  
1499854            NaN  
1499886            NaN  
1499902            0.0  
1499928            0.0  
1499979            NaN  

[19853 rows x 32 columns]

In [18]:
# There are five categories.  The third one is (-8.0, -1.0].  
pd.qcut(df.ARRIVAL_DELAY, 5)

0             (-15.0, -8.0]
1             (-15.0, -8.0]
2          (-81.001, -15.0]
3              (-1.0, 13.0]
4              (-8.0, -1.0]
                 ...       
1499995        (-1.0, 13.0]
1499996      (13.0, 1971.0]
1499997      (13.0, 1971.0]
1499998        (-8.0, -1.0]
1499999    (-81.001, -15.0]
Name: ARRIVAL_DELAY, Length: 1500000, dtype: category
Categories (5, interval[float64, right]): [(-81.001, -15.0] < (-15.0, -8.0] < (-8.0, -1.0] < (-1.0, 13.0] < (13.0, 1971.0]]

In [19]:
# or we can ask to list the third category.  
pd.qcut(df.ARRIVAL_DELAY, 5).cat.categories[2]

Interval(-8.0, -1.0, closed='right')

#### **Q7** How many of the origin airport codes in the flights dataset do not have a mapping to a real airport in the aiports file?  For example, some values of ORIGIN_AIRPORT are numeric, eg 14869.  These appear to be not valid airport codes per the aiports file.)
#### (Hint: There are many ways to solve this one.  Here is one approach: Left join the flights data (flights.csv) with the airports data (airports.csv) using the column ORIGIN_AIRPORT.  Then count the nulls in any of the new fields that are coming from the 'right' table, eg AIRPORT.)

In [20]:
# We can do the merge as described in the question, and count the sum of the isnull rows
df.merge(df_airports, how = 'left' , left_on = 'ORIGIN_AIRPORT', right_on = 'IATA_CODE').AIRPORT.isnull().sum()

125799

In [21]:
# Or we can just do the merge and look for the number of rows at the bottom
df[df.merge(df_airports, how = 'left' , left_on = 'ORIGIN_AIRPORT', right_on = 'IATA_CODE').AIRPORT.isnull()]

Unnamed: 0  YEAR  MONTH  DAY  DAY_OF_WEEK AIRLINE  FLIGHT_NUMBER  \
28          4665937  2015     10   18            7      OO           4469   
77          4511174  2015     10    8            4      DL           1383   
91          4762165  2015     10   24            6      HA              9   
113         4711425  2015     10   21            3      EV           4155   
134         4613388  2015     10   15            4      AA           2162   
...             ...   ...    ...  ...          ...     ...            ...   
1499967     4596532  2015     10   14            3      OO           5584   
1499971     4646625  2015     10   17            6      EV           4971   
1499976     4854840  2015     10   30            5      WN            401   
1499984     4551003  2015     10   11            7      B6            311   
1499987     4714099  2015     10   21            3      AS             50   

        TAIL_NUMBER ORIGIN_AIRPORT DESTINATION_AIRPORT  ...  ARRIVAL_TIME  \
28           N608SK          14831               14869  ...        2105.0   
77           N335NB          14122               10397  ...        2128.0   
91           N370HA          12892               12173  ...        2124.0   
113          N13913          12266               12448  ...        1559.0   
134          N959UW          12953               10721  ...        1201.0   
...             ...            ...                 ...  ...           ...   
1499967      N707SK          12266               11433  ...        1414.0   
1499971      N917EV          11433               11003  ...        1255.0   
1499976      N236WN          14730               13232  ...        1601.0   
1499984      N316JB          10721               13930  ...        1551.0   
1499987      N764AS          14709               10754  ...        1819.0   

         ARRIVAL_DELAY  DIVERTED  CANCELLED  CANCELLATION_REASON  \
28               -10.0         0          0                  NaN   
77                 8.0         0          0                  NaN   
91                24.0         0          0                  NaN   
113               -8.0         0          0                  NaN   
134              -14.0         0          0                  NaN   
...                ...       ...        ...                  ...   
1499967          -16.0         0          0                  NaN   
1499971            7.0         0          0                  NaN   
1499976          -14.0         0          0                  NaN   
1499984          -14.0         0          0                  NaN   
1499987            5.0         0          0                  NaN   

         AIR_SYSTEM_DELAY  SECURITY_DELAY  AIRLINE_DELAY  LATE_AIRCRAFT_DELAY  \
28                    NaN             NaN            NaN                  NaN   
77                    NaN             NaN            NaN                  NaN   
91                    0.0             0.0           20.0                  4.0   
113                   NaN             NaN            NaN                  NaN   
134                   NaN             NaN            NaN                  NaN   
...                   ...             ...            ...                  ...   
1499967               NaN             NaN            NaN                  NaN   
1499971               NaN             NaN            NaN                  NaN   
1499976               NaN             NaN            NaN                  NaN   
1499984               NaN             NaN            NaN                  NaN   
1499987               NaN             NaN            NaN                  NaN   

         WEATHER_DELAY  
28                 NaN  
77                 NaN  
91                 0.0  
113                NaN  
134                NaN  
...                ...  
1499967            NaN  
1499971            NaN  
1499976            NaN  
1499984            NaN  
1499987            NaN  

[125799 rows x 32 columns]

#### **Q8** In the flights dataset, what is the maximum count of missing values in a given row?

In [22]:
# We do a sum of isnulls for the rows, and sort ascending to see that 16 is the max number of values missing in a row
df.isnull().sum(axis=1).sort_values(ascending=False)

1295591    16
1350121    16
307417     16
1094020    16
879662     16
           ..
868188      1
233778      1
1218880     1
868189      1
681281      1
Length: 1500000, dtype: int64

#### **Q9** What is the average distance traveled by flights departing from the top 5 busiest origin airports (by number of departures)?

In [23]:
# First, let us see what the top-5 origin airports are
df.ORIGIN_AIRPORT.value_counts()

ORIGIN_AIRPORT
ATL      89592
ORD      73457
DFW      61573
DEN      50502
LAX      49912
         ...  
11503        2
15497        2
13502        2
13459        2
10165        1
Name: count, Length: 628, dtype: int64

In [24]:
# Let us take the top 5 of the above
df.ORIGIN_AIRPORT.value_counts()[:5].index

Index(['ATL', 'ORD', 'DFW', 'DEN', 'LAX'], dtype='object', name='ORIGIN_AIRPORT')

In [25]:
df[df.ORIGIN_AIRPORT.isin(df.ORIGIN_AIRPORT.value_counts()[:5].index)].DISTANCE.mean()

815.7857314266728